src/services/toros/easySwapper.ts

```diff
@@ -0,0 +1,109 @@
+import { ethers } from "ethers";
+import { Dapp, Pool } from "../..";
+
+import IDhedgeEasySwapper from "../../abi/IDhedgeEasySwapper.json";
+import { routerAddress } from "../../config";
+import { getChainlinkPriceInUsd } from "../chainLink/price";
+import { isTorosPool, loadTorosPool } from "./pool";
+
+export async function getPoolDepositAsset(
+  pool: Pool,
+  poolAddress: string
+): Promise<string> {
+  const torosPool = await loadTorosPool(pool, poolAddress);
+  const composition = await torosPool.getComposition();
+  return composition.find(e => e.isDeposit)?.asset as string;
```

Comment:
IMO, it shouldn't force casting to return `string` if `undefined` is also possible to be one of the returned values

Reply:
Undefined would only be returned if there is no deposit asset. When pools are set up at least one asset must be a deposit asset, so this case doesn't exist. How would you implement it to avoid the typescript error?

Reply:
that will be an edge case, it's fine if you're confident about that hardly happens.
for the typescript lint error, my thought is to change the return type to `Promise<string | undefined>`, and let the invoker handle both cases